api/handlers.go

```diff
@@ -260,18 +294,21 @@ func (api *API) registerPOST(w http.ResponseWriter, req *http.Request, _ httprou
 		api.WriteError(w, errors.AddContext(err, "failed to validate challenge response"), http.StatusBadRequest)
 		return
 	}
-	// Validate the email address.
-	email := req.PostFormValue("email")
-	e, err := mail.ParseAddress(email)
+	// Parse the request's body.
+	var payload credentialsDTO
+	err = json.Unmarshal(body, &payload)
+	if err != nil {
+		api.WriteError(w, errors.AddContext(err, "failed to parse request body"), http.StatusBadRequest)
+		return
+	}
+	// Validate and normalize the email address.
+	payload.Email, err = lib.NormalizeEmail(payload.Email)
```

Comment:
nit: probably worth a `UnmarshalPayload` method that combines these two actions so that callers don't forget one.

Reply:
Add a new type `EmailField` with a custom unmarshaller. I just need to write its tests and fix all the other tests that broke in the process.